### DISCUSSION TITLE
Silently fall back to regex highlighting when no parser?

### DISCUSSION THREAD

**Discussion by treatmesubj** [nvim-treesitter/nvim-treesitter#7394] [Q&A] [answered]
An error is raised by neovim when I open a filetype for which I do not have a tree-sitter parser installed, and I have no syntax highlighting.
Is this error expected, and can it be suppressed, and can Neovim fall back to the oldschool regex syntax highlighting in this case?
It seems that I should not have to install all parsers to avoid these errors.

My `init.lua` snippet:
```lua
require('nvim-treesitter.configs').setup {
  -- A list of parser names, or "all"
  ensure_installed = { "yaml", "json" },
  highlight = {
    enable = false,
    additional_vim_regex_highlighting = false,
  }
}
```

Error when opening a filetype for which I do not have the treesitter-parser installed:
```
Error detected while processing BufReadPost Autocommands for "*":
Error executing lua callback: /usr/share/nvim/runtime/filetype.lua:35: Error executing lua: /usr/share/nvim/runtime/
filetype.lua:36: BufReadPost Autocommands for "*"..FileType Autocommands for "*"..function <SNR>1_LoadFTPlugin[20]..
script /usr/share/nvim/runtime/ftplugin/lua.lua: Vim(runtime):E5113: Error while calling lua chunk: /usr/share/nvim/
runtime/lua/vim/treesitter/language.lua:107: no parser for 'lua' language, see :help treesitter-parsers
```

```
$ nvim --version
NVIM v0.10.2
Build type: Release
LuaJIT 2.1.1713484068
Run "nvim -V1 -v" for more info
```

**Comment by clason** [comment]
> An error is raised by neovim when I open a filetype for which I do not have a tree-sitter parser installed, and I have no syntax highlighting.
Is this error expected, 

No. If you don't have a `lua` parser installed, your installation is broken. (The parser is bundled with Neovim, same as other runtime files.)

**Reply by treatmesubj** [reply]
I've got [vim-plug](https://github.com/junegunn/vim-plug) with below in my `~/.vimrc`
```vim
Plug 'nvim-treesitter/nvim-treesitter', {'do': ':TSUpdate'}
```
and below in my `~/.init.lua`
```lua
vim.api.nvim_command('set runtimepath^=~/.vim runtimepath+=~/.vim/after')
vim.api.nvim_command('let &packpath = &runtimepath')
vim.api.nvim_command('source ~/.vimrc')
```
and after doing a `:TSInstall lua` and a `:echo nvim_get_runtime_file('parser/*.so', v:true)`, I see
```
['/home/john/.vim/plugged/nvim-treesitter/parser/json.so', '/home/john/.vim/plugged/nvim-treesitter/parser/lua.so',
'/home/john/.vim/plugged/nvim-treesitter/parser/yaml.so']
```
@clason any chance you know if this is an improper set-up in relation to `tree-sitter`? And do you have any tips?

**Reply by clason** [reply, marked as answer]
No, I literally mean that your Neovim installation is broken. This error should never appear, even if you don't have nvim-treesitter installed.

**Reply by treatmesubj** [reply]
ok thanks looks like I had a typo in my install command, I accidentally cleaned out `/usr/lib/nvim`
```bash
curl -L "https://github.com/neovim/neovim/releases/latest/download/nvim-linux64.tar.gz" \
    -o ~/nvim-linux64.tar.gz \
    && tar xzvf ~/nvim-linux64.tar.gz --directory ~ \
    && sudo rm -rf /usr/bin/nvim && sudo cp -r ~/nvim-linux64/bin/* /usr/bin \
    && sudo rm -rf /usr/lib/nvim && sudo cp -r ~/nvim-linux64/lib/* /usr/lib \
    && sudo rm -rf /usr/lib/nvim && sudo cp -r ~/nvim-linux64/share/* /usr/share \
    && rm -rf ~/nvim-linux64*
```

after fixing above command, `:echo nvim_get_runtime_file('parser/*.so', v:true)` looks more sensible now
```
['/home/john/.vim/plugged/nvim-treesitter/parser/json.so', '/home/john/.vim/plugged/nvim-treesitter/parser/yaml.so',
 '/usr/lib/nvim/parser/c.so', '/usr/lib/nvim/parser/lua.so', '/usr/lib/nvim/parser/markdown.so', '/usr/lib/nvim/pars
er/markdown_inline.so', '/usr/lib/nvim/parser/query.so', '/usr/lib/nvim/parser/vim.so', '/usr/lib/nvim/parser/vimdoc
.so']
```

**Reply by treatmesubj** [reply]
And I guess that error was only thrown by Neovim for lua filetype because it expects `/usr/lib/nvim/parser/lua.so` to be installed, as it is now bundled with Neovim, like you say.
